src/lib/hint_util.ts

```diff
@@ -0,0 +1,424 @@
+/**
+ * Helper types for :hint
+ */
+
+import Logger from "@src/lib/logging"
+import * as DOM from "@src/lib/dom"
+import * as hinting from "@src/content/hinting"
+
+/**
+ * Open mode: how to act on the selected hintable element
+ */
+export enum OpenMode {
```

Comment:
I've rebased this on master, where we added a new yarn command `yarn run rebuild` that uses `esbuild` and is much quicker (1 second versus 40 seconds).

It looks like `esbuild` can't deal with these `enum`s though - I get the error

`# controller in excmd: ReferenceError: can't access lexical declaration 'OpenMode' before initialization` whenever I try to run hint mode.

Do you think you might have time to look into it? We don't use `enum` anywhere else in the code base so if it looks like it's insurmountable could we rewrite the code to not use `enum`s?

We'll probably eventually switch to using `esbuild` for "proper" builds too as it really is much quicker.

Thanks : )

Reply:
8a059af fixes this. It seems esbuild miscompiles the `const` declaration I used to alias `hint_util.OpenMode` in `hint`, but enums are properly supported by esbuild!